tank_lab_to_nwb/convert_towers_task/virmenbehaviordatainterface.py

```diff
@@ -104,80 +112,88 @@ def convert_data(self, nwbfile: NWBFile, metadata_dict: dict, stub_test: bool =
-                                 for epoch in matin['log']['block'] for trial in epoch.trial]
-            right_cue_offset_data, _ = create_indexed_array(right_cue_offsets)
-
-            left_cue_positions = [trial.cuePos[0] if np.any(trial.cuePos[0]) else np.nan
-                                  for epoch in matin['log']['block'] for trial in epoch.trial]
-            left_cue_position_data, _ = create_indexed_array(left_cue_positions)
-
-            right_cue_positions = [trial.cuePos[1] if np.any(trial.cuePos[1]) else np.nan
-                                   for epoch in matin['log']['block'] for trial in epoch.trial]
-            right_cue_position_data, _ = create_indexed_array(right_cue_positions)
+            left_cue_onsets = [trial['time'][trial['cueOnset'][0] - 1]
+                               if np.any(trial['cueOnset'][0]) else np.nan for trial in trials]
+            trial_left_cue_onsets = [trial_start + cue_onset for trial_start, cue_onset
+                                     in zip(trial_starts, left_cue_onsets)]
+            left_cue_onset_data, left_cue_onset_indices = create_indexed_array(
+                trial_left_cue_onsets)
+
+            right_cue_onsets = [trial['time'][trial['cueOnset'][1] - 1]
+                                if np.any(trial['cueOnset'][1]) else np.nan for trial in trials]
+            trial_right_cue_onsets = [trial_start + cue_onset for trial_start, cue_onset
+                                      in zip(trial_starts, right_cue_onsets)]
+            right_cue_onset_data, right_cue_onset_indices = create_indexed_array(
+                trial_right_cue_onsets)
+
+            left_cue_offsets = [trial['time'][trial['cueOffset'][0] - 1]
+                                if np.any(trial['cueOffset'][0]) else np.nan for trial in trials]
+            trial_left_cue_offsets = [trial_start + cue_offset for trial_start, cue_offset
+                                      in zip(trial_starts, left_cue_offsets)]
+            left_cue_offset_data, left_cue_offset_indices = create_indexed_array(
+                trial_left_cue_offsets)
+
+            right_cue_offsets = [trial['time'][trial['cueOffset'][1] - 1]
+                                 if np.any(trial['cueOffset'][1]) else np.nan for trial in trials]
+            trial_right_cue_offsets = [trial_start + cue_offset for trial_start, cue_offset
+                                       in zip(trial_starts, right_cue_offsets)]
+            right_cue_offset_data, right_cue_offset_indices = create_indexed_array(
+                trial_right_cue_offsets)
+
+            left_cue_positions = [trial['cuePos'][0] if np.any(trial['cuePos'][0]) else np.nan
+                                  for trial in trials]
+            left_cue_position_data, left_cue_position_indices = create_indexed_array(
+                left_cue_positions)
+
+            right_cue_positions = [trial['cuePos'][1] if np.any(trial['cuePos'][1]) else np.nan
```

Comment:
Are you able to use an empty list instead of NaNs?

Reply:
Yeah, empty list would be better. For the first try when adding to nwb file I got this error: `could not resolve dtype for VectorData`, but I'm going to take a second look.

Reply:
@bendichter When trying to store empty array I get `OSError: Can't write data (no appropriate function for conversion path)`. There is a try-except block [here](
https://github.com/hdmf-dev/hdmf/blob/fab5660b1e009151980939e266e63a6c408064aa/src/hdmf/backends/hdf5/h5tools.py#L1362) that raises an exception when trying to assign data of mixed data types. I don't think this can be solved on our side, does it make sense to raise an issue at `hdmf-dev/hdmf`?

Reply:
@weiglszonja can you come up with a minimal example that demonstrates what we want to do and triggers this error?

Reply:
@bendichter This example triggers the same `OSError: Can't write data (no appropriate function for conversion path)` when trying to store data with empty array.  
```
import h5py
with h5py.File('my_hdf5_file.h5', 'w') as f:
    dset = f.create_dataset('test', (7,))
    data = [1, 2, 3, 4, [], 5, 6]
    dset[:] = data
```

Reply:
@weiglszonja yes but you won't actually be storing an empty list when this is converted to a pynwb indexed array. @luiztauffer or @CodyCBakerPhD, can you help here?